include/bitcoin/explorer/define.hpp

```diff
@@ -104,7 +104,8 @@ enum class encoding_engine
 {
     info,
     json,
-    xml
+    xml,
+    data
```

Comment:
I'm not current enough to cite our precedent, but "data" is otherwise ambiguous. It would seem that "binary" is a more specific moniker, thoughts?

Reply:
This will require changes to all commands with encodings, their tests, wiki help documentation and wiki examples.

Reply:
Data, or binary doesn't matter as much to me -- but it's not really binary since it's base16 encoded, which makes it misleading too.  Preference?

As for changes to all commands, it seems the encodings are mostly used for distinguishing if output is json or not in the prop_tree (if used at all in the commands), and otherwise just left to the callback_state's output methods.  It looks like base16 encoding based on the engine can be entirely handled in the output methods such as:

https://github.com/libbitcoin/libbitcoin-explorer/blob/master/src/callback_state.cpp#L67

Thoughts?

Reply:
Yes, def better to do it within the callback handler. IIRC JSON was a late addition, but now we'll need an enumerated parameter, which is better. We also still need true JSON, maybe boost has it by now? We could use "hex" as a token for this. I've tried to avoid it, in favor of the explicit "base16" in most areas. But in this case I think it would be better and I don't believe it creates any inconsistency.

Reply:
Doing this would require us to overload `output(...)` to accept various data types along with the format enumeration. IMO that pushes the formatting too far down. So as with all computer science problems, this is resolved with another layer of abstraction. We should pass a stream ref and enumeration into `output`, and then use a class/fn that accepts the data types and serializes them using `property_tree` or base16, etc. as the data parameter to `output`.

Reply:
This is overcome by incorporation of boost::json serialization for all objects and planned changes to explorer parameterization.